Source/Core/DolphinQt2/HotkeyScheduler.cpp

```diff
@@ -0,0 +1,350 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/HotkeyScheduler.h"
+
+#include <algorithm>
+#include <thread>
+
+#include <QCoreApplication>
+
+#include "AudioCommon/AudioCommon.h"
+#include "Common/Thread.h"
+#include "Core/Core.h"
+#include "Core/HotkeyManager.h"
+#include "Core/IOS/IOS.h"
+#include "Core/IOS/USB/Bluetooth/BTBase.h"
+#include "Core/State.h"
+#include "DolphinQt2/MainWindow.h"
+#include "DolphinQt2/Settings.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "VideoCommon/VertexShaderManager.h"
+#include "VideoCommon/VideoConfig.h"
+
+HotkeyScheduler::HotkeyScheduler(MainWindow* parent) : m_parent(parent)
+{
+  HotkeyManagerEmu::Initialize();
+  HotkeyManagerEmu::LoadConfig();
+  HotkeyManagerEmu::Enable(true);
+}
+
+void HotkeyScheduler::Start()
+{
+  m_stop_requested = false;
+  m_thread = std::thread(&HotkeyScheduler::Run, this);
+  m_thread.detach();
```

Comment:
There's no point in storing a reference _and_ detaching.

Probably what you want is to store the (still-attached) reference here and then `join()` it in the destructor.